kolibri_explore_plugin/viewsets.py

```diff
@@ -35,11 +64,62 @@ def _consolidate(self, items, queryset):
             else:
                 tags[t["tagged_content"]].append(t["tag_name"])
 
+        # We need to batch our queries for ancestors as the size of
+        # the expression tree depends on the number of nodes that we
+        # are querying for.  On Windows, the SQL parameter limit is
+        # 999, and an ancestors call can produce 3 parameters per node
+        # in the queryset, so this should max out the parameters at
+        # 750.
+        ANCESTOR_BATCH_SIZE = 250
+
+        if len(items) > ANCESTOR_BATCH_SIZE:
+
+            ancestors_map = {}
+
+            for i in range(0, len(items), ANCESTOR_BATCH_SIZE):
+
+                for anc in (
+                    ContentNode.objects.filter(
+                        id__in=[
+                            item["id"]
+                            for item in items[i : i + ANCESTOR_BATCH_SIZE]
+                        ]
+                    )
+                    .get_ancestors()
+                    .values("id", "title", "lft", "rght", "tree_id")
+                ):
+                    ancestors_map[anc["id"]] = anc
+
+            ancestors = sorted(ancestors_map.values(), key=lambda x: x["lft"])
+        else:
+            ancestors = list(
+                queryset.get_ancestors()
+                .values("id", "title", "lft", "rght", "tree_id")
+                .order_by("lft")
+            )
+
+        def ancestor_lookup(item):
```

Comment:
If the call to `super().consolidate` works correctly, the item has the `ancestors` property but not the `lft` and `rght` properties.

https://github.com/learningequality/kolibri/blob/release-v0.14.x/kolibri/core/content/api.py#L375

```suggestion
        def ancestor_lookup(item):
            if "ancestors" in item:
                return item

```